resource-managers/mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosCoarseGrainedSchedulerBackend.scala

```diff
@@ -380,7 +389,8 @@ private[spark] class MesosCoarseGrainedSchedulerBackend(
       } else {
         declineOffer(
           driver,
-          offer)
+          offer,
```

Comment:
As an aside, what do you think about adding more detailed logging of the reason why offers are declined?

Reply:
I agree ok.

Reply:
I've found the `reason` to give less information then desired..

Reply:
At this point we decline any offer we couldnt use to launch a task... In order to have a reason like why the checkcanLaunchTask previously failed for that offer we have to store the related info somewhere. Alternatively I propose to log that failure at the canLaunchTask call earlier and call decline here as follows:
```
        declineOffer(
          driver,
          Some("Offer was declined due to unmet task launch constraints.")
          offer)
```
Sounds good?

Reply:
Yeah for now that's fine. I was thinking more specific messages but we can address that on a later ticket.

Reply:
There is a debug message if canLaunchTask fails. I will create a jira and improve logging across the module.

Reply:
That is fixed.